Add first tags for Caracal

Alex-Welsh opened the pull request.

Alex-Welsh committed:
Add first tags for Caracal

markgoddard commented on etc/kayobe/ansible/prometheus.yml.j2:
```diff
@@ -247,18 +247,6 @@ scrape_configs:
 {% endfor %}
 {% endif %}
 
-{% if enable_ironic_prometheus_exporter | bool %}
```
Weird - this is in stable/2024.1 but not stable/2023.1

Alex-Welsh replied:
There are legitimate issues in this file but they're not relevant to getting CI passing. I've dropped those changes for now and will come back to them

markgoddard commented on etc/kayobe/stackhpc.yml:
```diff
@@ -145,7 +145,8 @@ stackhpc_kolla_source_version: "stackhpc/{{ openstack_release }}"
 stackhpc_kolla_ansible_source_url: "https://github.com/stackhpc/kolla-ansible"
-stackhpc_kolla_ansible_source_version: "stackhpc/{{ openstack_release }}"
+# stackhpc_kolla_ansible_source_version: "stackhpc/{{ openstack_release }}"
+stackhpc_kolla_ansible_source_version: "fix-prometheus"
```
Looks like this fix needs to go upstream.

Alex-Welsh replied:
https://review.opendev.org/c/openstack/kolla-ansible/+/926198

markgoddard replied:
Always good to add a FIXME for these things.

Alex-Welsh replied:
done

cityofships replied:
Ready to revert?

Alex-Welsh replied:
Yeah, think so

markgoddard commented:
Check tags fails due to heat being disabled. Need to add `kolla_enable_heat: true` in ci-builder environment, then rebuild heat images.

markgoddard commented:
Upgrade jobs fail due to use of classic RMQ queues. Need to migrate to quorum queues before upgrade in CI job.

markgoddard committed:
Caracal: Bump releasenotes version RE

markgoddard committed:
docs: Fix extlinks for updated sphinx

The slug parameter is now required in the link f-string.

Alex-Welsh committed:
Allow reboot playbook to run with bootstrap user

Alex-Welsh committed:
Add first tags for Caracal Heat

Alex-Welsh committed:
Work around RL9 OVN hostname bug in CI

OVN Deployments on Rocky 9 cannot create VMs when hostnames contain a
dot `.` which includes CI AIOs.

This commit removes the .novalocal suffix from the AIO hostname and
reboots the VM to apply changes before proceeding with the deployment.

Alex-Welsh committed:
Work around upstream kolla-ansibe prometheus bug

Alex-Welsh committed:
Fix releasenotes

Alex-Welsh commented:
To prove the reno fixes work, I applied them [here](https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/10353005199) but they'll need a force-push first

Alex-Welsh commented:
TODOs:
Get kolla-ansible prometheus fix pushed upstream
Merge https://github.com/stackhpc/stackhpc-kayobe-config/pull/1221 to fix upgrade jobs
Fix OVN hostname issue upstream
Update prometheus config templates
Merge all the other PRs
Rebuild containers for OVN bump
Upgrade RabbitMQ one version at a time

MoteHue commented on .github/workflows/stackhpc-all-in-one.yml:
```diff
@@ -344,6 +354,17 @@ jobs:
           KAYOBE_AUTOMATION_SSH_PRIVATE_KEY: ${{ steps.ssh_key.outputs.ssh_key }}
         if: inputs.upgrade
 
+      - name: Migrate RabbitMQ queues
```
Given this will be running from the Caracal checkout, this will also cause the new version of the RabbitMQ-using services to be deployed, but without any of the handling KA usually does to make the upgrade smooth :grimacing:. 
Ideally, you should migrate the queues when still on Antelope.

Alex-Welsh replied:
Okay, I'll slice this out and put it in Antelope

Alex-Welsh replied:
Done (untested while waiting on https://github.com/stackhpc/stackhpc-kayobe-config/pull/1224/files)

MoteHue commented on etc/kayobe/ansible/migrate-rabbitmq-queues.yml:
```diff
@@ -0,0 +1,84 @@
+    - name: Migrate RabbitMQ queues
+      shell:
+        cmd: >
+          kayobe overcloud service configuration generate --node-config-dir /etc/kolla --kolla-skip-tags rabbitmq-ha-precheck -ke om_enable_rabbitmq_quorum_queues=true -ke om_enable_rabbitmq_high_availability=false &&
```
If we intend for this playbook to be used outside of CI (we should, but it should go into Antelope), we'll want to make sure this gets persisted in globals.yml. Just in case someone runs the queue migration, but then doesn't immediately upgrade to Caracal.

MoteHue commented on etc/kayobe/ansible/migrate-rabbitmq-queues.yml:
```diff
@@ -0,0 +1,84 @@
+---
```
Instead of using ``-ke om_enable_rabbitmq_quorum_queues=true -ke om_enable_rabbitmq_high_availability=false`` and then the debug message at the end, can we add an assert to the start of the playbook that makes sure these are set? This way we enforce that it's set in config.

Alex-Welsh replied:
Done, see https://github.com/stackhpc/stackhpc-kayobe-config/pull/1224/files

MoteHue commented on releasenotes/notes/rabbit-disable-ha-9b150bcc9e4d4f54.yaml:
```diff
@@ -0,0 +1,10 @@
+---
+features:
+  - |
+    RabbitMQ will now use quorum queues by default, rather than traditional HA.
```
"traditional HA" isn't really the term we've used in the past, and I don't think there's any merit in saying it tbh.
```suggestion
    RabbitMQ will now use quorum queues by default.
```

Alex-Welsh replied:
Removed, see https://github.com/stackhpc/stackhpc-kayobe-config/pull/1224/files

MoteHue reviewed (changes requested):
Loving the queue migration playbook, but I have comments :)

MoteHue commented on etc/kayobe/ansible/migrate-rabbitmq-queues.yml:
```diff
@@ -0,0 +1,84 @@
+        cmd: >
+          kayobe overcloud service configuration generate --node-config-dir /etc/kolla --kolla-skip-tags rabbitmq-ha-precheck -ke om_enable_rabbitmq_quorum_queues=true -ke om_enable_rabbitmq_high_availability=false &&
+          kayobe kolla ansible run "stop --yes-i-really-really-mean-it" -kt {{ services_to_restart }} -ke om_enable_rabbitmq_quorum_queues=true -ke om_enable_rabbitmq_high_availability=false &&
+          kayobe kolla ansible run rabbitmq-reset-state -ke om_enable_rabbitmq_quorum_queues=true -ke om_enable_rabbitmq_high_availability=false &&
```
We had some checks [here](https://stackhpc-kayobe-config.readthedocs.io/en/stackhpc-2023.1/operations/rabbitmq.html) that made sure all OpenStack queues and exchanges have been removed from the RabbitMQ cluster before starting the services. This caught cases where a service might have been missed from the stop (e.g. a hypervisor removed from an inventory, but still running services). Would be good to get those in here:

```
kayobe overcloud host command run --limit controllers --show-output --command 'docker exec rabbitmq rabbitmqctl list_queues name'
kayobe overcloud host command run --limit controllers --show-output --command 'docker exec rabbitmq rabbitmqctl list_exchanges name'
```
There should be no queues listed, and the only exchanges listed should start with amq..

Alex-Welsh replied:
Rabbit migration work has been moved to another PR

Alex-Welsh committed:
Use Quorum queues in Rabbit

markgoddard commented on .github/workflows/stackhpc-all-in-one.yml:
```diff
@@ -264,6 +264,16 @@ jobs:
         run: |
           docker image pull $KAYOBE_IMAGE
 
+      - name: Reboot
```
Can you add a comment describing why we're rebooting here?

There are some related PRs [here](https://github.com/stackhpc/stackhpc-kayobe-config/pull/1088/files) and [here](https://github.com/stackhpc/stackhpc-kayobe-config/pull/1144/files).

Alex-Welsh replied:
Done

markgoddard commented on .github/workflows/stackhpc-all-in-one.yml:
```diff
@@ -304,6 +314,24 @@ jobs:
+            -v $(pwd):/stack/kayobe-automation-env/src/kayobe-config \
+            -e KAYOBE_ENVIRONMENT -e KAYOBE_VAULT_PASSWORD -e KAYOBE_AUTOMATION_SSH_PRIVATE_KEY \
+            ${{ steps.kayobe_image.outputs.kayobe_image }} \
+            /stack/kayobe-automation-env/src/kayobe-config/.automation/pipeline/playbook-run.sh etc/kayobe/ansible/migrate-rabbitmq-queues.yml --tags rabbit-queue-migration
```
Do we need a kayobe-automation script to run an SKC script?

Alex-Welsh replied:
Unfortunately yes I think so. I was putting it off by doing things in playbooks cause the scope keeps growing but I don't really see a way around it now

markgoddard replied:
I think it would be generally useful.

Alex-Welsh replied:
WIP PR here: https://github.com/stackhpc/kayobe-automation/pull/62/files
Testing here: https://github.com/stackhpc/stackhpc-kayobe-config/pull/1233/

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/ansible/reboot.yml:
```diff
@@ -2,6 +2,12 @@
 - name: Reboot the host
   hosts: seed-hypervisor:seed:overcloud:infra-vms
   serial: "{{ lookup('env', 'ANSIBLE_SERIAL') | default(1, true) }}"
+  vars:
```
Need to sync 2023.1

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/environments/ci-aio/inventory/group_vars/cis-hardening/cis:
```diff
@@ -0,0 +1,3 @@
+---
+# Don't use FIPS since it blocks ed25519 keys
+rhel9cis_crypto_policy: DEFAULT
```
Wait for 2023.1 discussion.

Alex-Welsh replied:
Done

markgoddard commented on terraform/aio/vm.tf:
```diff
@@ -92,7 +92,8 @@ resource "null_resource" "kayobe-aio" {
       "#!/bin/sh",
-      "echo 'connected!'"
+      "echo 'connected!'",
+      "sudo hostnamectl set-hostname $(sed s/.novalocal// /etc/hostname)"
```
This could use an explanatory comment.

Alex-Welsh replied:
Done

Alex-Welsh committed:
Add explanatory comments for Caracal changes

Alex-Welsh committed:
Merge remote-tracking branch 'origin/stackhpc/2023.1' into caracal-aios

Alex-Welsh committed:
All-in-one workflow fixes for Caracal

markgoddard commented on .github/workflows/stackhpc-all-in-one.yml:
```diff
@@ -329,15 +329,16 @@ jobs:
             -e KAYOBE_ENVIRONMENT -e KAYOBE_VAULT_PASSWORD -e KAYOBE_AUTOMATION_SSH_PRIVATE_KEY \
             ${{ steps.kayobe_image.outputs.kayobe_image }} \
-            /stack/kayobe-automation-env/src/kayobe-config/.automation/pipeline/playbook-run.sh etc/kayobe/ansible/migrate-rabbitmq-queues.yml --tags rabbit-queue-migration
+            /stack/kayobe-automation-env/src/kayobe-config/.automation/pipeline/script-run.sh tools/rabbitmq-quorum-migration.sh --skip-checks
```
Skip checks? :(

Alex-Welsh replied:
Well the checks used to fail you see :slightly_smiling_face:

Alex-Welsh replied:
I think they'll pass now. I'll remove the skip once the rabbit changes merge in Antelope so I can actually test the changes here

Alex-Welsh committed:
Bump .automation checkout

This changes is required for the new script-run.sh
tool, which is used in the AIO upgrade CI job to
migrate RabbitMQ queues

Alex-Welsh committed:
Enable rabbit upgrade checks

Alex-Welsh committed:
Merge remote-tracking branch 'origin/stackhpc/2024.1' into caracal-aios

markgoddard commented:
Do we need to close/reopen the PR to show the correct commits?

Alex-Welsh commented:
> Do we need to close/reopen the PR to show the correct commits?

Doesn't seem to have helped. I've proposed a sync [here](https://github.com/stackhpc/stackhpc-kayobe-config/pull/1235/files) which should clear it up a bit

Alex-Welsh committed:
Merge branch 'stackhpc/2024.1' into caracal-aios

Alex-Welsh commented:
> > Do we need to close/reopen the PR to show the correct commits?
> 
> Doesn't seem to have helped. I've proposed a sync [here](https://github.com/stackhpc/stackhpc-kayobe-config/pull/1235/files) which should clear it up a bit

Much better :smile:

Alex-Welsh committed:
Make fix-hostname playbook more generic

Alex-Welsh committed:
Update ovn hostname bug workaround

Alex-Welsh commented:
@markgoddard all good now!
